spec.bs

```diff
@@ -4107,6 +4160,68 @@ A worklet function is one of the following:
 
 </dl>
 
+<div algorithm>
+To <dfn>find corresponding bid and score phase function</dfn> given a [=worklet function=] |fn|:
+
+  1.Switch on on |fn|:
+    <dl class="switch">
+    : [=worklet function/generate-bid=]
+    :: Return [=worklet function/generate-bid=].
+    : [=worklet function/score-ad=]
+    :: Return [=worklet function/score-ad=].
+    : [=worklet function/report-result=]
+    :: Return [=worklet function/score-ad=].
+    : [=worklet function/report-win=]
+    :: Return [=worklet function/generate-bid=].
+
+    </dl>
+</div>
+
+### Averager ### {#private-aggregation-averager}
+An <dfn>averager</dfn> is a a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="averager">
+  : <dfn>count</dfn>
+  :: A {{long}}, initially 0.
+  : <dfn>sum</dfn>
+  :: A {{double}}, initially 0.
+</dl>
+
+<div algorithm>
+To <dfn>add a sample to an averager</dfn> given an [=averager=] |averager|, a {{double}} |sample|:
+1. Increment |averager|'s [=averager/count=] by 1.
+1. Increment |averager|'s [=averager/sum=] by |sample|.
+
+</div>
+
+<div algorithm>
+To <dfn>merge samples to an averager</dfn> given [=averagers=] |dest| and |source|:
+1. Set |dest|'s [=averager/count=] to |dest|'s [=averager/count=] + |source|'s [=averager/count=].
+1. Set |dest|'s [=averager/sum=] to |dest|'s [=averager/sum=] + |source|'s [=averager/sum=].
+
+</div>
+
+<div algorithm>
+To <dfn>get the value to report from an averager</dfn> given an [=averager=] |averager|:
+1. If |averager|'s [=averager/count=] is 0, return 0.
+1. Return |averager|'s [=averager/sum=] / [=averager/count=].
+
+</div>
+
+### Metrics structures ### {#private-aggregation-metrics-structures}
+An <dfn>execution metrics</dfn> is a [=struct=] with the following [=struct/items=]:

```

Comment:
would be nice to have a brief explaination about what it holds, like "is a struct which has per-ig metrics ...". Similar to the other struct below, "... which has per-participant metrics that's only reported once per-participant rather than for every interest group participating, with the ...."

Reply:
Done.